### ISSUE TITLE
Widget shows wrong IP adress while using VPN

### ISSUE THREAD

**Issue by Nivram42** [Davide-sd/ip_address#11]
Hello, thanks for this cool widget!
As long as no VPN connection is established, everything including the VPN indicator works just fine. 
But when I connect to a VPN, the widget keeps showing the old IP address. 
And when I remove the widget and add a new one, it shows no information until I disconnect the VPN connection.  
If you need logs, please tell me what you need and how i can get them. 
-----------------------------------------
Manjaro Linux 
KDE Plasma 5.18.5
KDE Framework 5.70.0
QT Version 5.14.2
Kernel 5.4.40-1

**Comment by Davide-sd** [comment]
> But when I connect to a VPN, the widget keeps showing the old IP address.

I believe this is expected, because I still have to connect the VPN and IP address together... Right now they are based on 2 different timers... So, IP address is updated every X minutes (you can set X in the settings). If you activate/deactivate the VPN, it will take up to X minutes to update the IP address. In the meantime, you can press the Update button after you connect/disconnect the VPN (I'll synchronize the two functionalities later on, once I have a decent confident on the VPN status check)...

> And when I remove the widget and add a new one, it shows no information until I disconnect the VPN connection.

That's even more strange... Does it shows the IP address? Did you let it "wait" for the X minutes I mentioned above? Let me know.

In the meantime, thanks for taking the time to report this issue :)

**Comment by Nivram42** [comment]
I have tried it again to be sure. 
First run:
-Update timer is set to 2 min
-VPN connection is established
-Wait for 3 minutes
-> nothing happens, the displayed IP and location are still the old ones 
-click on the update button
-> still nothing changed

Second Run:
-VPN is still connected
-remove the widget
-wait 3 minutes
-set a fresh one in the control bar
-the widget shows no values, just "N/A"
-click on Update changed nothing

![Screenshot_20200527_202121](https://user-images.githubusercontent.com/66025718/83057787-accdf880-a057-11ea-85b5-ae53d3e13861.png)

-> immediately after disconnecting the VPN and clicking the Update button, the widget shows the correct information.

It makes no difference if the widget is bound on the control bar or "flying" on the desktop.
Without VPN everything is works like it should.

I am planing to do a new clean system install in the next days. Maybe it will work on a fresh system

**Comment by Davide-sd** [comment]
Yep, I've been able to replicate the issue. There is a bug in the update check. Gonna fix it soon, hang tight (next few days...).

**Comment by Nivram42** [comment]
Hi,
I have noticed you have changed the code on github. The Version on the KDE Store seems to be the older version. 
I have tried to install the github version by copying the content from 
/p_address-master/plasmoid/ 
to
 ~/.local/share/plasma/plasmoids/com.github.davide-sd.ipaddress/ 
and run kpackagetool5 --generate-index -t Plasma/Applet. 

After that I can see the new Logo in the "add a new widget menu" but I can not Drag and Drop it in the status bar or the desktop. 

Am I doing it completely wrong? 

And i have one feature suggestion: the option to disable the default widget icon, so that only the VPN notification is displayed (when activated).

**Comment by atassis** [comment]
You can track which network has a priority to be used via `ip route show` command. The lowest metric will chow which device has the most priority.

**Comment by Davide-sd** [comment]
Solved it. To deal with the requests, I was using `XMLHttpRequest`, which is unable to deal with network changes. Now I'm using another method. I will release a new version on Pling in a few days.
